api.go

```diff
@@ -178,6 +178,11 @@ func (c *Client) getJSONWithRetry(
 	method, uri string,
 	body, resp interface{}) error {
 
+	if c == nil || c.api == nil {
```

Comment:
Do we really need this condition? if the client is nil. When we call this method from somewhere. There itself it panics with nil pointer dereference.